contents/staticAnalysis/FindBugs.md

```diff
@@ -0,0 +1,59 @@
+
+FindBugs analyses code based on compiled Java `.class` file and checks multiple files at the same time. These feature overcomes some limitations in [CheckStyle](CheckStyle.md) and [PMD](PMD.md). Therefore, FindBugs can spot potential errors that CheckStyle and PMD can not find. For example, one of the bug patterns in FindBugs is `RCN: Redundant nullcheck of value known to be non-null`. Only after a comprehensive analysis of all the assignments to a variable in the code base, we can check whether the `nullcheck` of value is redundant or not.
+
+## How to use it
+
+### Configuration
+
+You can tell FindBugs which bug pattern to exclude and which bug pattern to include by using [FilterFiles](http://findbugs.sourceforge.net/manual/filter.html). By default, if not filter files is provided, FindBugs will run all checks.
+
+### Suppress Warnings
+
+You can use the filter file with `exclude` options in FindBugs as discussed above to suppress warnings.
+
+In addition, you can also use `SuppressWarnings` [annotation](http://findbugs.sourceforge.net/manual/annotations.html) to filter out unwanted violations.
+
+### Running
+
+There are several ways to run FindBugs.
+
+GUI:
+
+- [Using the FindBugs GUI](http://findbugs.sourceforge.net/manual/gui.html)
+
+Command Line:
+
+- [Command Line](http://findbugs.sourceforge.net/manual/running.html)
+
+Build Automation Tools:
+
+- [Ant Task](http://findbugs.sourceforge.net/manual/anttask.html)
+- [FindBugs Maven Plugin](http://gleclaire.github.io/findbugs-maven-plugin/)
+- [Gradle FindBugs](https://docs.gradle.org/current/userguide/findbugs_plugin.html)
+
+IDE Integration:
+
+- [Eclipse Integration](http://findbugs.sourceforge.net/manual/eclipse.html)
+- [IntelliJ Integration](https://plugins.jetbrains.com/plugin/3847-findbugs-idea)
+- [NetBeans Integration](https://netbeans.org/kb/docs/java/code-inspect.html)
+
+## Advanced Topic
+
+- [Data mining of bugs with FindBugs](http://findbugs.sourceforge.net/manual/datamining.html): The data for each analysis will be collected and you can use these statistics for data mining purpose.
+- [Configure Analysis Properties](http://findbugs.sourceforge.net/manual/analysisprops.html#analysisproptable): You can define properties to configure options in some checks. For example, you can define the assertion methods in your project so that null pointer dereference bug detector will not raise violations if assertion methods are used.
+- [Use annotations](http://findbugs.sourceforge.net/manual/annotations.html): You can use annotation to indicated your intent so that FindBugs can issue warnings more appropriately.
+
+## Resources
+
+- [An Evaluation of FindBugs](http://www.cs.cmu.edu/~aldrich/courses/654/tools/Sandcastle-FindBugs-2009.pdf): Analysis of FindBugs in 2009, some content may be outdated. It is good to looks at the benefits and drawbacks of FindBugs.
```

Comment:
... may be outdated. **Useful for understanding** the benefits and drawbacks...

_Also would be good if there's link to a basic tutorial. Especially since your only resource might be 'outdated'. :P_